enterprise/server/backends/userdb/userdb.go

```diff
@@ -554,17 +554,24 @@ func (d *UserDB) RequestToJoinGroup(ctx context.Context, userID string, groupID
 	})
 }
 
-func (d *UserDB) GetGroupUsers(ctx context.Context, groupID string, statuses []grpb.GroupMembershipStatus) ([]*grpb.GetGroupUsersResponse_GroupUser, error) {
-	if err := perms.AuthorizeGroupAccess(ctx, d.env, groupID); err != nil {
+func (d *UserDB) GetGroupUsers(ctx context.Context, statuses []grpb.GroupMembershipStatus) ([]*grpb.GetGroupUsersResponse_GroupUser, error) {
+	u, err := perms.AuthenticatedUser(ctx, d.env)
+	if err != nil {
 		return nil, err
 	}
+	if err := authutil.AuthorizeGroupRole(u, u.GetGroupID(), role.Admin); err != nil {
+		return nil, err
+	}
+	if len(statuses) == 0 {
+		return nil, status.InvalidArgumentError("statuses list must not be empty")
+	}
```

Comment:
It's nice for readability (and performance) to check preconditions before other stuff -- can we move this up before L558?

Reply:
Done